src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private Object[] array;
+    private int size;
+    
+    public ArrayList() {
+        this.array = new Object[INITIAL_SIZE];
+    }
+    
     @Override
     public void add(T value) {
-
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
+        array[size++] = value;
     }
-
+    
     @Override
     public void add(T value, int index) {
-
+        if (size != index) {
+            checkIndex(index);
+        }
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
-
+    
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
-
+    
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) array[index];
     }
-
+    
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
-
+    
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T element = (T) array[index];
+        System.arraycopy(array, index + 1, array, index, size - index);
+        array[size--] = null;
+        return element;
     }
-
+    
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
+            if (element == array[i] || element != null && element.equals(array[i])) {
+                System.arraycopy(array, i + 1, array, i, size - i);
+                array[size--] = null;
+                return element;
+            }
+        }
+        throw new NoSuchElementException("element is absant value : " + element.toString());
```

Comment:
```suggestion
        throw new NoSuchElementException("element is absent value : " + element);
```
toString is called by java